### PR TITLE
Go back v0 as default

### PR DESCRIPTION
**Pull request by gosusnp**
## What
Addresses complications with upgrading the platform and normalization to support V1 protocol changes. This change will focus on making sure that platform is downgrading V1 protocols on-the-fly to ensure the catalog stored is V0. This effort will need to be in coordination with the normalization being downgraded to support V0 protocol

## How
Adjust changes that were made to update the protocol version to V1 to go back and support V0 protocol until the migration process can be handled 

## Recommended reading order
1. `x.java`
2. `y.python`

### CODE REVIEW

File: airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/migrations/v1/CatalogMigrationV1Helper.java [hunk 1]
```diff
@@ -118,4 +118,102 @@ private static boolean hasV0DataType(final JsonNode schema) {
...
+
+  private static boolean streamContainsV1DataTypes(final AirbyteStream airbyteStream) {
+    if (airbyteStream == null || airbyteStream.getJsonSchema() == null) {
+      return false;
+    }
+    return hasV1DataType(airbyteStream.getJsonSchema());
+  }
+
+  /**
+   * Performs of search of a v0 data type node, returns true at the first node found.
```
Comment (edgao) [inline airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/migrations/v1/CatalogMigrationV1Helper.java:204]:
```suggestion
   * Performs of search of a v1 data type node, returns true at the first node found.
```

**Review by ryankfu — APPROVED** [review]
LGTM

**Review by edgao — APPROVED** [review]
one nit on a comment. Code changes look right. I don't feel qualified to say if there were any files that should have been changed, that were missed though.

**Review by colesnodgrass — APPROVED** [review]
Changes seem fine.  What's the confidence that the downgrade code works as expected?

**Comment by octavia-approvington** [comment]
A crack team of mammals has made a decision.  <br/> imagine a seal of approval

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**